content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+
+The Rails framework is guided by two major principles:
+
+1. Don’t Repeat Yourself: The [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) is very common in software development. The idea is that developers should spend less time writing code by avoiding repetition. This is done by making sure every element of your code can stand alone and be called anywhere and changes to that element don't have to be repeated across several points.
+2. Convention over Configuration: This is one of Rails’ superpowers. Rails has a set of guidelines and format for doing things. From naming methods, classes, files, and building a database schema. These conventions help to keep the whole application in sync and help you spend less time debugging errors that result from the wrong configuration.
+
+## Setting up your dev environment: Installing Ruby, Rails, and PostgreSQL
+
+[GoRails](https://goRails.com/) Has in-depth guides on setting up Rails on various machines. Here's a link to all of them. Since we're using PostgreSQL for this project, skip any step that mentions setting up SQLite or MySQL.
+
+- [For Ubuntu](https://goRails.com/setup/ubuntu/22.04)
+- [Mac OS](https://goRails.com/setup/macos/13-ventura)
+- [For Windows](https://www.hanselman.com/blog/ruby-on-Rails-on-windows-is-not-just-possible-its-fabulous-using-wsl2-and-vs-code)
+
+## Setting up a new Rails app
+
+To create a new Rails app, simply open up your terminal, navigate to where you want this app to stay and run the following command.
+
+```bash
+rails new todolist -d postgresql –css tailwind
+```
+
+The commands after the app name (`todolist`) are optional. The `-d postgresql` flag tells rails to use PostgreSQL as the database for this app. The `–css tailwind` flag tells rails to use tailwind as the CSS framework for this app. You can use any CSS framework you want or just plain CSS. I chose tailwind because it’s easy to use and it’s a breeze to set up.
+
+After running the command, you’ll see a bunch of files and folders being created. This is the default structure of a Rails app. You can read more about it [here](https://guides.rubyonrails.org/getting_started.html#creating-the-blog-application).
+
+To get our app running, we need to make sure it is connected to a database. The database config lives in the `config/database.yml` file. There we can see the provided database name and some other info. This database doesn't exist yet and needs to be created. For most people, you need to supply your PostgreSQL username and password that was set during installation to this file. The default username is `postgres` and this applies to most Linux distros.
+
+Most times we would use an env file to manage this but we can keep it there for the scope of this project.
+
+```yml
+...
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  username: postgres
+  password: yourpassword
+...
+```
+
+To create the database, run `rails db:create`.
```

Comment:
Both commands work for the same purpose. `rake db:create` is the legacy way of creating a new database. 

`rails db:create` is the more preferred way because it uses the Rails application framework and the underlying database adapter to create the database instead of the Rake task runner

Reply:
I'm using Rails 7.0.4.2 on Ubuntu and `rails db:create`  didn't work. It just gave me the rails help menu. Is there some reason it wouldn't work? Should I be in a particular directory?

 I'm not a rails expert and I'm just trying to follow along with the tutorial.

Reply:
Please make sure your PostgreSQL service is running before trying that command.

`sudo service postgresql start`

Reply:
That seems to work. That detail should be in the article.